wgpu-core/src/command/render.rs

```diff
@@ -1172,6 +1176,7 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
                     state.index.bound_buffer_view = Some((id::Valid(buffer_id), offset..end));
                     state.index.update_limit();
 
+                    state.index.format = index_format;
```

Comment:
We actually need to track both the `Option<IndexFormat>` from the current pipeline and the current index format of the active `SetIndexBuffer`. Before issuing a draw call there is an `is_ready` function that needs to check for:
  - if the pipeline index format is set, it has to be the same as the current index format, otherwise an error is generated